core/src/main/scala/quasar/destination/avalanche/AvalancheConfig.scala

```diff
@@ -79,20 +80,28 @@ object AvalancheConfig {
       ad => (ad.clientId.value, ad.tenantId.value, ad.clientSecret.value).some)(
       "clientId", "tenantId", "clientSecret")
 
-  implicit val avalancheConfigCodecJson: CodecJson[AvalancheConfig] =
-    casecodec5[String, String, URI, String, AzureCredentials.ActiveDirectory, AvalancheConfig](
-      (accountName, containerName, uri, password, creds) =>
-        AvalancheConfig(
-          AccountName(accountName),
-          ContainerName(containerName),
-          uri,
-          ClusterPassword(password),
-          creds),
-      cfg =>
-        (cfg.accountName.value,
-          cfg.containerName.value,
-          cfg.connectionUri,
-          cfg.password.value,
-          cfg.azureCredentials).some)(
-      "accountName", "containerName", "connectionUri", "clusterPassword", "credentials")
+  implicit def AvalancheConfigCodecJson: CodecJson[AvalancheConfig] =
+    CodecJson({ (c: AvalancheConfig) =>
+        ("accountName" := c.accountName.value) ->:
+        ("containerName" := c.containerName.value) ->:
+        ("connectionUri" := c.connectionUri) ->:
+        ("clusterPassword" := c.password.value) ->:
+        c.writeMode.map("writeMode" := _) ->?:
```

Comment:
This codec change is entirely for this line.

Reply:
Which I chose to keep this roundtrip compatible as that seems safest.

Reply:
:+1: on backwards compatibility.

:golf: `c.writeMode.map("writeMode" := _)` == `("writeMode" :?= c.writeMode)`